### ISSUE TITLE
doctest with cabal-doctest and stack on Windows

### ISSUE THREAD

**Issue by kazu-yamamoto** [kazu-yamamoto/unix-time#47]
I'm trying to let `doctest` work with `cabal-doctest`. Now `cabal test` and `stack test` work well on my macOS machine.

Since `unix-time` now supports Windows, I tried it on Windows but failed:

https://ci.appveyor.com/project/kazu-yamamoto/unix-time/builds/21110057

```
ghc.exe:  | C:\stack\.stack-work\dist\7d103d30\build\HSunix-time-0.4.2-6iaC9OYeAn5BldWGBIKrw.o: unknown symbol `__imp_timezone'
```
A Window user said to me that this also happens  in `stack repl`:

```
ghc.EXE: | C:\Users\kazukiokamoto\Projects\unix-time\.stack-work\dist\7d103d30\build\cbits\strptime.o: unknown symbol `__imp_tzname'
```

Both `__imp_timezone` and `__imp_tzname` exist in `libmsvcrt.a` of Mingw.

@snoyberg I guess that `stack` does not specify Mingw CRT-related libraries in the case of `stack test` (for `doctest`) and `stack repl`. Is this a bug of `stack` or are there any ways to resolve this issue?

Cc: @kakkun61 and @sighingnow

**Comment by snoyberg** [comment]
IIRC, this is some kind of a bug with the GHCi linker and name resolution on Windows. I remember battling issues like this in the past. I think I worked around it previously by adding dummy wrappers in my own C files in the project and referring to those wrappers instead.

**Comment by sighingnow** [comment]
I have encountered different doctest failure on windows:

```
λ stack test
unix-time-0.4.2: test (suite: doctest)


Data\UnixTime\Conv.hs:14:1: error:
    Could not find module ‘Data.UnixTime.Types’
    it is a hidden module in the package ‘unix-time-0.4.2’
    Use -v to see a list of the files searched for.
   |
14 | import Data.UnixTime.Types
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

unix-time-0.4.2: Test suite doctest failed
unix-time-0.4.2: test (suite: spec)
```

**Comment by sighingnow** [comment]
Fixed in #48 .

**Comment by kazu-yamamoto** [comment]
@snoyberg Thank you for your answer.
@sighingnow You are great.

Let's close this!
